connectors/file/translator-parquet/src/main/java/org/teiid/translator/parquet/BaseParquetExecution.java

```diff
@@ -0,0 +1,338 @@
+    private MessageType schema;
+    private MessageType filteredSchema;
+    private RecordReader<Group> rowIterator;
+    private MessageColumnIO columnIO;
+    private long pageRowCount;
+    protected HashMap<String, String> partitionedColumnsValue;
+    protected HashMap<String, Integer> partitionedColumnsHm = new HashMap<>();
+
+    public BaseParquetExecution(ExecutionContext executionContext,
+                                RuntimeMetadata metadata, VirtualFileConnection connection, boolean immutable) {
+        this.executionContext = executionContext;
+        this.metadata = metadata;
+        this.connection = connection;
+        this.immutable = immutable;
+    }
+
+    public void visit(LanguageObject command) throws TranslatorException {
+        this.visitor.visitNode(command);
+
+        if (!visitor.exceptions.isEmpty()) {
+            throw visitor.exceptions.get(0);
+        }
+    }
+
+    @Override
+    public void execute() throws TranslatorException {
+        String path = this.visitor.getParquetPath();
+        if(this.visitor.getTable().getProperty(ParquetMetadataProcessor.PARTITIONED_COLUMNS) != null) {
+            path += getDirectoryPath(this.visitor.getColumnPredicates(), this.visitor.getTable().getProperty(ParquetMetadataProcessor.PARTITIONED_COLUMNS));
+        }
+        this.parquetFiles = VirtualFileConnection.Util.getFiles(path, this.connection, true, false);
+        VirtualFile nextParquetFile = getNextParquetFile();
+        if (nextParquetFile != null) {
+            this.rowIterator = readParquetFile(nextParquetFile);
+        }
+    }
+
+    private String getDirectoryPath(HashMap<String, Comparison> predicates, String partitionedColumns) throws TranslatorException {
+        StringBuilder path = new StringBuilder(); // because we are only supporting the equality comparison as of now, otherwise we would return a list of paths to iterate over
+        String[] partitionedColumnsArray = getPartitionedColumns(partitionedColumns);
+        for (String s : partitionedColumnsArray) {
+            partitionedColumnsHm.put(s, 1);
+            if (predicates.containsKey(s)) {
+                Comparison predicate = predicates.get(s);
+                predicates.remove(s);
+                Literal l = (Literal) predicate.getRightExpression();
+                path.append("/").append(s).append("=");
+                if (predicate.getOperator() == Comparison.Operator.EQ) {
+                    path.append(l.getValue().toString());
+                }
+            } else {
+                path.append("/*");
+            }
+        }
+        path.append("/*");
+    return path.toString();
+    }
+
+    private FilterCompat.Filter getRowGroupFilter(HashMap<String, Comparison> columnPredicates) throws TranslatorException {
+        if(columnPredicates.size() == 0){
+            return FilterCompat.NOOP;
+        }
+        Iterator hashMapIterator = columnPredicates.entrySet().iterator();
```

Comment:
Please use generic types when possible.  For example instead of just Iterator, use Iterator<Entry<String, Comparison>> - the ide should have an option to infer generic types as a formatting/cleanup task.